### DISCUSSION TITLE
Way to make saving all of your maps between computers easier

### DISCUSSION THREAD

**Discussion by Glitch102x** [ppy/osu#17492] [Ideas] [answered]
I have quite the buggy computer, so I have had to do this twice now. The best options I've thought up or found are to either screenshot all the map names in the songs folder to download later, or to save them on a cloud drive (which often didn't work due to slow internet speeds and said buggy internet). 

Now, this is bad. The first time, I honestly debated just dropping the game. Of course, I was doomed to remain here eternally by then, but I was seriously debating. Losing everything you've spent months building up is not a great sensation. So, since we're giving osu!direct for free in Lazer (I think, correct me if I'm wrong), why don't we let players save the maps they have? It wouldn't even be really memory intensive if it's backed up on client-side servers based on account, but it'll be even cheaper on that front if we allowed the option to just create a relatively small file that contained the information on all the maps. Instead of saving the maps themselves, it would just contain which maps have to be downloaded by direct and then execute that. Much easier to back up on a drive folder. 

Do forgive me if this has been suggested before, pepper man. My search skills are not very great.

**Comment by bdach** [comment, marked as answer]
https://github.com/ppy/osu/issues/2351

**Comment by peppy** [comment]
You can already view all beatmaps you've played in the past: https://osu.ppy.sh/beatmapsets?played=played

You could also just back up to a drive as you seem to be against. Trust that this saves for everyone (you and our servers) and would be faster than re-downloading gigabytes for each user that can't be bothered using a flash drive for some reason.

**Reply by Glitch102x** [reply]
That, unfortunately, requires a supporter tag to do. It would also be a little tedious to go through and click download on every map, but that is not as big of a complaint from me. And exporting such large file sizes to drives with my internet speeds and computer stability is just not very plausible.

I apologize for my lack of understanding of how the servers work and any misunderstandings about how plausible it is to make the game do something, but I was more referring to just making the game print out a file containing the link to the beatmap page on the website. I'm pretty sure the 'quick web access for beatmap' button in atleast stable should contain some information on the link. So, if the game prints out the instruction file just on the information the player's folder already contains, I don't think it would take up any server space at all (atleast on your end). A file like this could be backed up on my end a lot easier than the gigabytes of maps I have.

Again, I'm mostly just bullshiting most of this. It may be entirely impractical or impossible to code this in. It's just a little annoying to backup my song list with the tools I have, so I'm spitballing with the very limited knowledge I have.

**Reply by bdach** [reply]
what good would a "list of maps" do you if you lose the maps? you'd still presumably want to _get the maps back to play them_, no? at which point we're back to "downloading gigabytes" land any way you cut it, and those gigabytes have to come either from a cloud drive/USB stick, or from the game's servers which costs money. if everyone starts to download their entire map collections on the fly from the client, we're starting to talk about a level of traffic that is just not sustainable.

**Reply by peppy** [reply]
> And exporting such large file sizes to drives with my internet speeds and computer stability is just not very plausible.

The proposal was *not* using the internet. Presuming you are in possession of both PCs. Might be a foreign concept to people these days, but computes can communicate without using the internet, if they are in the same vicinity.

**Reply by Glitch102x** [reply]
Tht does make sense, Bdach. I don't have any fixes to that particular problem there
